lib/livebook/runtime/standalone_init.ex

```diff
@@ -5,15 +5,20 @@ defmodule Livebook.Runtime.StandaloneInit do
   # a new Elixir system process. It's used by both ElixirStandalone
   # and MixStandalone runtimes.
 
-  alias Livebook.Utils
   alias Livebook.Utils.Emitter
+  alias Livebook.Runtime.NodePool
 
   @doc """
   Returns a random name for a dynamically spawned node.
   """
   @spec child_node_name(atom()) :: atom()
   def child_node_name(parent) do
-    :"#{Utils.random_short_id()}-#{parent}"
+    if Process.whereis(NodePool) do
+      NodePool.get_name(parent)
+    else
+      NodePool.start()
```

Comment:
Instead of starting it on demand, you want to add it to your supervision tree in lib/livebook/application.ex. you can list it as a child like this:

    Livebook.Runtime.NodePool

And then add this to nodepool:

    def start_link(opts) do
      buffer_time = opts[:buffer_time] || @default_time
      name = opts[:name] || __MODULE__
      GenServer.start_link(__MODULE__, buffer_time, name: name)
    end

Then in your tests, you can do this:

    start_supervised!(NodePool, name: config.test, buffer_time: ...)

Reply:
Note you also no longer need the start/* functions.